pages/send/Manual.tsx

```diff
@@ -1,49 +1,160 @@
+import { LN_ADDRESS_REGEX } from "@getalby/lightning-tools";
+import { router } from "expo-router";
 import React, { useState } from "react";
-import { View } from "react-native";
+import { Pressable, ScrollView, View } from "react-native";
 import DismissableKeyboardView from "~/components/DismissableKeyboardView";
+import Loading from "~/components/Loading";
 import Screen from "~/components/Screen";
 import { Button } from "~/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardTitle,
+} from "~/components/ui/card";
+import { Checkbox } from "~/components/ui/checkbox";
 import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
 import { Text } from "~/components/ui/text";
 import { errorToast } from "~/lib/errorToast";
 import { initiatePaymentFlow } from "~/lib/initiatePaymentFlow";
+import { useAppStore } from "~/lib/state/appStore";
 
 export function Manual() {
   const [keyboardText, setKeyboardText] = useState("");
+  const [contactName, setContactName] = useState("");
+  const [addToContacts, setAddToContacts] = React.useState(false);
+  const addressBookEntries = useAppStore((store) => store.addressBookEntries);
+  const [isSubmitting, setSubmitting] = React.useState(false);
 
   const submitKeyboardText = async () => {
+    setSubmitting(true);
     try {
+      if (addToContacts) {
+        useAppStore.getState().addAddressBookEntry({
+          name: contactName,
+          lightningAddress: keyboardText,
+        });
+      }
```

Comment:
Should we do this only if `initiatePaymentFlow` returns true? Else if the user keeps on pressing "Next" for an invalid address it keeps on adding contacts

Reply:
good catch, added